tests/Functional/Cases/Rendering/DataAccessorTest.php

```diff
@@ -0,0 +1,207 @@
+<?php
+namespace TYPO3Fluid\Fluid\Tests\Functional\Cases\Rendering;
+
+use TYPO3Fluid\Fluid\Core\Cache\FluidCacheInterface;
+use TYPO3Fluid\Fluid\Core\Cache\SimpleFileCache;
+use TYPO3Fluid\Fluid\Tests\Functional\Cases\Rendering\Fixtures as Fixtures;
+use TYPO3Fluid\Fluid\Tests\UnitTestCase;
+use TYPO3Fluid\Fluid\View\TemplateView;
+
+class DataAccessorTest extends UnitTestCase
+{
+    /**
+     * @return array
+     */
+    public function dataIsRenderedDataProvider()
+    {
+        return [
+            // array values
+            [
+                ['value' => 'value'],
+                ['value' => 'value'],
+            ],
+            [
+                new \ArrayObject(['value' => 'value']),
+                ['value' => 'value'],
+            ],
+            // accessing using properties directly
+            [
+                $this->createObjectWithProperties(),
+                [
+                    'privateValue' => null,
+                    'protectedValue' => null,
+                    'publicValue' => 'publicValue',
+                ],
+                'Cannot access private property class@anonymous::$privateValue', // @todo Probably regression?!
+            ],
+            [
+                $this->createObjectWithProperties(),
+                [
+                    'publicValue' => 'publicValue',
+                ],
+            ],
+            // accessing using getters
+            [
+                $this->createObjectWithGetter(),
+                [
+                    'privateValue' => 'privateValue@getPrivateValue()',
+                    'protectedValue' => 'protectedValue@getProtectedValue()',
+                    'publicValue' => 'publicValue@getPublicValue()', // @todo Probably regression?!
+                ],
+            ],
+            // accessing using magic method __call()
+            [
+                $this->createObjectWithMagicCall(),
+                [
+                    'privateValue' => null,
+                    'protectedValue' => null,
+                    'publicValue' => 'publicValue@__call("getPublicValue")',
+                ],
+                'Cannot access private property class@anonymous::$privateValue' // @todo Probably regression?!
```

Comment:
This is the pain point. If the property exists, public or not, the `__call` method is not consulted. See https://github.com/TYPO3/Fluid/issues/483 - TL;DR we likely can't do this without Reflection and I *really* don't want to start using Reflection for extracting properties. Issue has three workarounds; `__call` support was introduced recently and has no truly established convention (my recommendation is to avoid it).